### PR TITLE
TLS Feature

### PR DESCRIPTION
**Pull request by ghost**
- Adding TLS support and certificate validations in current MQTT implementation
- Adding identity layer to encapsulate gateway identity and TLS configurations

### CODE REVIEW

File: liota/transports/mqtt.py [hunk 1]
```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
...
+                else:
+                    raise ValueError("Error : Wrong client key path.")
+            else:
+                client_key_available = False
+
+            # Multiple conditions for certificate validations
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If one of them is not there throw an error to notify one of them is missing
+            if client_cert_available != client_key_available:
```
Comment (diwanc) [inline liota/transports/mqtt.py]: 1) Boolean operator "not" preferable over "!="
2) Also, think of splitting the first case into two; a) "Client cert file found. But key not found." b) "Client key file found. But cert file not found."

File: liota/transports/mqtt.py [hunk 2]
```diff
@@ -48,25 +49,78 @@ def on_message(self, client, userdata, msg):
...
+            else:
+                self.client.tls_set(self.gw_identity.cacert,
+                                    cert_reqs=getattr(ssl, self.TLSConf.cert_required),
+                                    tls_version=getattr(ssl, self.TLSConf.tls_version), ciphers=self.TLSConf.cipher)
+
+            log.info("TLS support is set up.")
 
+        # Connect with MQTT Broker
+        self.client.on_connect = self.on_connect
+        self.client.connect(host=self.url, port=self.port, keepalive=60)
```
Comment (diwanc) [inline liota/transports/mqtt.py]: Is keepalive not a param to be collected at method level? If yes, lets do that and have a default value.

File: liota/transports/mqtt.py (hunk 1 above)
Comment (ghost) [inline liota/transports/mqtt.py]: Sure. We can do this

File: liota/transports/mqtt.py (hunk 2 above)
Comment (ghost) [inline liota/transports/mqtt.py]: Ok. We can add keepalive as configurable param with value 60 in conf file

File: liota/transports/mqtt.py [hunk 3]
```diff
@@ -48,25 +49,82 @@ def on_message(self, client, userdata, msg):
...
+                else:
+                    raise ValueError("Error : Wrong client key path.")
+            else:
+                client_key_available = False
+
+            # Multiple conditions for certificate validations
+                # 1. Both Client certificate and key file should be present
+                # 2. If both are not there proceed without client certificate and key
+                # 3. If client certificate is not there throw an error
+                # 4. If client key is not there throw an error
```
Comment (diwanc) [inline liota/transports/mqtt.py:102]: This is more elegant.

**Pull request merged**